bayes_kit/rhat.py

```diff
@@ -15,7 +92,7 @@ def rhat(chains: list[SeqType]) -> FloatType:
     `phi[m] = mean(chains[m])` and `psi[m] = var(chains[m])`.  This reduces to
     the standard definition when all chains are the same length.
```

Comment:
My bad.  I meant to write `chains` where I wrote `theta` (the variable `theta` is what's used in all the original papers and doc).  

I removed "standard" and just cited the paper and noted that the paper only defines rhat for rectangular structures whereas this definition generalizes to ragged structures in a way that matches the original for rectangular inputs.

Is it that you don't know the `mean` function or that it's confusing that `N` is an array or that we're talking about the mean of a sample rather than a random variable?  I clarified all three points, so it's now extra pedantic. I'm sticking to `N` because that matches the paper we're implementing and the books that define it.   I also clarified we're using sample mean and sample variance (as opposed to random variable mean and variance) in case that was confusing you and supplied the NumPy functions to compute the exact values (hope that's not too close to the code).  I'd strongly prefer not to define sample mean and variance in the definition of rhat.

The actual interpretation of the value is very tricky.  It's about the scale (aka std deviation) of the values in the chains.  (So it's not the scale of the current estimate that's reduced, but the scale of the estimator---estimators are random variables and thus have a standard deviation---usually the sample standard deviation of an estimator is called the "standard error").

I can't cite BDA3, because the definition there is different and could be confusing, and it's also bad form to cite secondary sources.  I do cite BDA3 and the Stan Reference Manual for split R-hat as that's where it was first described.